node/p2p/routing.go

```diff
@@ -8,12 +8,21 @@ import (
 	"github.com/libp2p/go-libp2p-core/protocol"
 	"github.com/libp2p/go-libp2p-core/routing"
 	dht "github.com/libp2p/go-libp2p-kad-dht"
+	routinghelpers "github.com/libp2p/go-libp2p-routing-helpers"
 	"go.uber.org/fx"
 
 	"github.com/celestiaorg/celestia-node/node/fxutil"
 )
 
-func Routing(cfg *Config) func(routingParams) (routing.PeerRouting, error) {
+// ContentRouting constructs nil content routing,
+// as for our use-case existing ContentRouting mechanisms, e.g DHT, are unsuitable
+func ContentRouting() routing.ContentRouting {
+	return &routinghelpers.Null{}
+}
```

Comment:
is this necessary to have then?

Reply:
Unfortunately yes, but I decide to move this out:
* To explicitly show that we don't rely on ContentRoting
* I feel like in a year from now we will come with something else and there will be some non-nil mechanism

Reply:
> There might be other places where we would need to fill it as some IPFS component depends on it, so we keeping it in FX make sense

Can we just introduce it when we absolutely need it? Or do you envision that we will basically need this immediately?

Reply:
Not immediately, but it does not hurt to keep it explicit, like now

Reply:
I'd rather not have unused code in our codebase in anticipation of something -- we will likely remove BitSwap soon / after Devnet, right? Will it be necessary beyond that point?

Reply:
It is in use, so it's not unused ;)

Reply:
By fx 😂  Okay if you feel so strongly that it should be there, then that's fine, but pls document it and maybe write a note about how it's only necessary as long as we use this IPFS component for data exchange

Reply:
> we will likely remove BitSwap soon / after Devnet, right?

Kinda. It is not clear whats the alternative. Retrospectively, I think we were very optimistic about designing our protocol for that. It makes sense to have something that would take lesser roundtrips, but we did not take into account that BitSwap also solves the problem of fetching data from multiple sources simultaneously, and reimplementing this is not trivial unless we fork it or something. So I wouldn't rely on any assumptions regarding BitSwap removal unless we defined the alternative.

Reply:
> By fx joy Okay if you feel so strongly that it should be there, then that's fine, but pls document it and maybe write a note about how it's only necessary as long as we use this IPFS component for data exchange

Wait, I thing you missed something. There is clearly one use of it, it is passed to BitSwap already.

My message:
> There might be other places where we would need to fill it as some IPFS component depends on it, so we keeping it in FX make sense

is about possibility of >1 usage, and there is clear =1 usage now. Look at the DataExchange(BitSwap) constructor

Reply:
Ah okay no problem then, let's just document it and mention that it's necessary for BitSwap. We should also probably document the bitswap portion to say that we will move away from it after Devnet.

Reply:
Pls, read my comment above regarding BitSwap removal. There is no point to document that, we already have that info in ADR